test_profilehooks.py

```diff
@@ -180,12 +180,9 @@ def doctest_coverage_when_source_is_not_available(self):
 
 
         >>> run_exitfuncs()
+
+        >>> print(buffer.getvalue().strip())
```

Comment:
You're not using `buffer` anywhere, so it'll always be blank.  What is the point of this check?